Add openscap plugins packages

ares opened the pull request:
I'll add more openscap related packages. Also we're missing ffi in EL7 so I have scratch for FC19 and EL6 only so far.
rubygem-openscap
http://koji.katello.org/koji/taskinfo?taskID=212754 FC19
http://koji.katello.org/koji/taskinfo?taskID=212787 EL6

ffi 1.4.0 scratch
http://koji.katello.org/koji/taskinfo?taskID=213172 EL7
http://koji.katello.org/koji/taskinfo?taskID=213173 EL6
http://koji.katello.org/koji/taskinfo?taskID=213175 FC19

rubygem-scaptimony scratch
http://koji.katello.org/koji/taskinfo?taskID=213185 EL7
http://koji.katello.org/koji/taskinfo?taskID=213187 EL6
http://koji.katello.org/koji/taskinfo?taskID=213183 FC19

rubygem-foreman_openscap
http://koji.katello.org/koji/taskinfo?taskID=213191 EL7
http://koji.katello.org/koji/taskinfo?taskID=213189 EL6
http://koji.katello.org/koji/taskinfo?taskID=213193 FC19

rubygem-smart_proxy_openscap
http://koji.katello.org/koji/taskinfo?taskID=213248 EL7
http://koji.katello.org/koji/taskinfo?taskID=213247 EL6
http://koji.katello.org/koji/taskinfo?taskID=213245 FC19

I didn't add rubygem-ffi to tito.props yet since I'm not sure where it's being built for EL6 and FC19. Is it a good idea that we 'SCL-ize' and build from foreman-packaging after it becoming one of our deps?

domcleal commented on rubygem-ffi/rubygem-ffi.spec:
```diff
@@ -1,25 +1,42 @@
 License:        LGPLv3
 URL:            http://wiki.github.com/ffi/ffi
-Source0:	http://rubygems.org/gems/%{gem_name}-%{version}.gem
+Source0:	      http://rubygems.org/gems/%{gem_name}-%{version}.gem
```
indentation's even further off now :)

domcleal commented on rubygem-ffi/rubygem-ffi.spec:
```diff
@@ -1,25 +1,42 @@
+Name:           %{?scl_prefix}rubygem-%{gem_name}
 Version:        1.4.0
-Release:        2%{?dist}
+Release:        3%{?dist}
```
either keep it at '2' and we'll run tito tag which'll bump it, or bump to '3' and add a %changelog

domcleal commented on rel-eng/tito.props:
```diff
@@ -338,6 +346,7 @@ whitelist = rubygem-algebrick
   rubygem-dynflow
   rubygem-jgrep
   rubygem-opennebula
+  rubygem-openscap
```
rubygem-openscap's in Fedora 19 I think

ares replied:
according to http://www.isitfedoraruby.com/fedorarpms/rubygem-openscap 20+ only

domcleal replied:
Ah yes, it's not actually been submitted.. it's just in Koji.

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,118 @@
+
+%global mainver 0.2.0
+%global release 1
+%{?prever:
```
just kill this section, not needed

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,118 @@
+%global gem_name scaptimony
+
+%global mainver 0.2.0
+%global release 1
```
if you can put these inline, it'd be appreciated (works better with tito)

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,118 @@
+%if 0%{?fedora} > 18
+Requires: %{?scl_prefix}ruby(release)
+Requires: %{?scl_prefix}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(deface)
```
no deface dependency

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,118 @@
+Requires: %{?scl_prefix}ruby(release)
+Requires: %{?scl_prefix}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(deface)
+Requires: %{?scl_prefix}rubygem(openscap) >= 0.4.1
```
0.4.0

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,118 @@
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+%if 0%{?fedora} > 18
+Requires: %{?scl_prefix}ruby(release)
```
missing a Requires on `rails ~> 3.2.8` (gah, @unorthodoxgeek will hate that!)

unorthodoxgeek replied:
I think I'm glad I don't understand this :smile: 
explain?

domcleal replied:
Just another thing that depends on a specific Rails version, but now I see it's a full blown engine, so I'm not surprised.

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,118 @@
+%else
+Requires: %{?scl_prefix}ruby(abi) >= %{rubyabi}
+Requires: %{?scl_prefix}rubygems
+Requires: %{?scl_prefix}rubygem-deface
```
move these (and those above) outside of the if/else

ares replied:
`rubygem-openscap` vs `rubygem(openscap)`, I guess the later would work for both?

domcleal replied:
Yeah, the latter will work (and is generally preferred) as the spec provides it: https://github.com/theforeman/foreman-packaging/pull/499/files#diff-e68b725840c22175ddfd40010237a90bR51

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+
+%global mainver 0.2.0
+%global release 1
+%{?prever:
```
same comment as scaptimony

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+Requires: %{?scl_prefix}ruby(release)
+Requires: %{?scl_prefix}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(deface)
+Requires: %{?scl_prefix}rubygem(scaptimony) >= 0.2.0
```
this dependency should be in the foreman_openscap gemspec, could you file/fix it there too please?

ares replied:
https://github.com/OpenSCAP/foreman_openscap/issues/67

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+%endif
+BuildRequires: %{?scl_prefix}rubygems-devel
+BuildRequires: %{?scl_prefix}rubygems
+BuildRequires: foreman-assets
```
`>= 1.7.0`

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+mkdir -p %{buildroot}%{foreman_bundlerd_dir}
+cat <<GEMFILE > %{buildroot}%{foreman_bundlerd_dir}/%{gem_name}.rb
+gem '%{gem_name}'
+gem 'scaptimony'
```
a symptom of the dep missing, ugh

ares replied:
I'll add #TODO comment and will remove when https://github.com/OpenSCAP/foreman_openscap/issues/67 is fixed

domcleal replied:
Thanks, it can be replaced by one line once fixed: `%foreman_bundlerd_file`

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+GEMFILE
+
+
+# Run the test suite
```
let's drop this no-op

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+        %{buildroot}%{gem_dir}/
+
+mkdir -p %{buildroot}%{foreman_bundlerd_dir}
+cat <<GEMFILE > %{buildroot}%{foreman_bundlerd_dir}/%{gem_name}.rb
```
%{buildroot}%{foreman_bundlerd_file}

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,137 @@
+%{gem_instdir}/config
+%exclude %{gem_cache}
+%{gem_spec}
+%{foreman_bundlerd_dir}/%{gem_name}.rb
```
%{buildroot}%{foreman_bundlerd_file}

ares replied:
are you sure with buildroot here?

domcleal replied:
Sorry, copy and paste error.  No buildroot, just the latter bit.

domcleal commented on rel-eng/tito.props:
```diff
@@ -374,9 +384,12 @@ whitelist = rubygem-algebrick
   rubygem-puppetdb_foreman
   rubygem-satyr
+  rubygem-scaptimony
+  rubygem-smart_proxy_openscap
```
duplicate of 392

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+%global gem_name smart_proxy_openscap
+
+%global foreman_proxy_bundlerd_dir /usr/share/foreman-proxy/bundler.d
```
/usr/share = %{_datadir}

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+%global gem_name smart_proxy_openscap
+
+%global foreman_proxy_bundlerd_dir /usr/share/foreman-proxy/bundler.d
+%global foreman_proxy_pluginconf_dir /etc/foreman-proxy/settings.d
```
/etc = %{_sysconfdir}

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+
+%global foreman_proxy_bundlerd_dir /usr/share/foreman-proxy/bundler.d
+%global foreman_proxy_pluginconf_dir /etc/foreman-proxy/settings.d
+%global spool_dir /var/spool/foreman-proxy/openscap
```
%{_var}

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+Release: 1%{?dist}
+Summary: OpenSCAP plug-in for Foreman's smart-proxy.
+Group: Applications/Internet
+License: GPLv2+
```
v3+

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+License: GPLv2+
+URL: http://github.com/openscap/smart_proxy_openscap
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+#Requires: ruby(release)
```
should be present, but behind a conditional for api/release

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+#Requires: ruby(release)
+Requires: ruby(rubygems)
+Requires: foreman-proxy >= 1.7.0-0.develop.201410221520
```
any idea what that's about?  It should refer to a release really

ares replied:
@isimluk that may be question for you ^

domcleal replied:
don't fret about it, I thought about it again and it doesn't really matter

isimluk replied:
Yep, that's pre 1.7.0 release. Written in a way that works even post release.

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+Requires: ruby(rubygems)
+Requires: foreman-proxy >= 1.7.0-0.develop.201410221520
+Requires: crontabs
+#BuildRequires: ruby(release)
```
ditto

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+BuildRequires: ruby
+BuildArch: noarch
+Provides: rubygem(%{gem_name}) = %{version}
+Obsoletes: rubygem-foreman-proxy_openscap
```
best to put a version in too, to avoid problems later if it changes, e.g. `Obsoletes: foo < 0.3.0-1`

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+       %{buildroot}%{gem_dir}/
+mv %{buildroot}%{gem_instdir}/%{gem_name}.gemspec %{buildroot}/%{gem_spec}
```
I don't think this is right, it should be using the gem_spec generated by the gem build above, not the original.

ares replied:
which should be done by cp %{gem_dir} already, right?

domcleal replied:
exactly.  I've a feeling it might be covering up a problem or workaround, but is likely just unused/pointless code.

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+# sample config
+mkdir -p %{buildroot}%{foreman_proxy_pluginconf_dir}
+mv %{buildroot}%{gem_instdir}/settings.d/openscap.yml.example \
+   %{buildroot}%{foreman_proxy_pluginconf_dir}/
```
install it as openscap.yml, then mark as %config(noreplace) in %files

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,101 @@
+%doc %{foreman_proxy_pluginconf_dir}/openscap.yml.example
+%config(noreplace) %attr(0644, root, root) %{_sysconfdir}/cron.d/%{name}
+
+%{gem_docdir}
```
a -doc package is preferred

domcleal commented:
Looks pretty good!  A couple of other comments:

1. add everything to comps/*plugin* (you can add the base RPM and run ./comps_doc.sh to update the -docs section)
2. foreman_scap is missing asset precompilation, just add `%foreman_precompile_plugin` to %install after the bundler.d file is created, and %{foreman_assets_plugin} to %files

ares commented:
@domcleal I fixed most of issues, could you go through new commits before I squash them please?

domcleal commented on rubygem-openscap/rubygem-openscap.spec:
```diff
@@ -0,0 +1,153 @@
+# For the tests we need
+BuildRequires: %{?scl_prefix}rubygem(test-unit)
+%else
+Requires:      %{?scl_prefix}ruby(abi) >= %{rubyabi}
```
rubyabi's not defined.  (I wouldn't bother, just remove the version spec.)

domcleal commented on rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec:
```diff
@@ -0,0 +1,112 @@
+%files doc
+%{gem_docdir}
+%{gem_instdir}/README.md
+%{gem_instdir}/COPYING
```
keep COPYING in the main RPM

domcleal commented on rubygem-scaptimony/rubygem-scaptimony.spec:
```diff
@@ -0,0 +1,109 @@
+Requires: %{?scl_prefix}ruby(release)
+Requires: %{?scl_prefix}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(openscap) >= 0.4.0
+Requires: %{?scl_prefix}rubygem(rails) >= 3.2.8
```
and < 3.3.0

ares replied:
fixed on line below

domcleal replied:
Seems to be in one if/else block, and not the other?  (Maybe just de-dupe it and move outside the conditional.)

domcleal commented on rubygem-foreman_openscap/rubygem-foreman_openscap.spec:
```diff
@@ -0,0 +1,123 @@
+
+mkdir -p %{buildroot}%{foreman_bundlerd_dir}
+# TODO: remove gem 'scaptimony' when https://github.com/OpenSCAP/foreman_openscap/issues/67 is fixed
+# this whole block can be replaced then by just %foreman_bundlerd_file
```
avoid mentioning the macro with %, or it'll expand inside here :)

domcleal commented:
Don't forget comps/ too.

ares commented:
I'm leaving this one for final commit. I'll have to add one more package (foreman_scap_client).

domcleal commented:
Okay, tests will fail until comps are updated as Jenkins doesn't know which tito build target to use.

ares commented:
Oh, that explains why it didn't try plugins target. Thanks for clarification.

domcleal commented:
Yeah, it was about the only way I could think to do it, as we don't really have a definitive list of what lives where..

shlomizadok commented:
@ares https://github.com/ares/foreman-packaging/blob/rpm/openscap/rubygem-foreman_openscap/rubygem-foreman_openscap.spec should be copied to OpenSCAP/foreman_openscap#31 ?

isimluk commented:
Nope, we should remove spec files from scap* upstreams once this is merged.

ares commented:
I added foreman_scap_client package
http://koji.katello.org/koji/taskinfo?taskID=213666 EL7
http://koji.katello.org/koji/taskinfo?taskID=213668 EL6
http://koji.katello.org/koji/taskinfo?taskID=213669 FC19

also I added comps changes, now looking at test failures

domcleal commented on rubygem-foreman_scap_client/foreman_scap_client.spec:
```diff
@@ -0,0 +1,90 @@
+%global gem_name foreman_scap_client
+%global config_dir %{_sysconfdir}/%{gem_name}
+
+Name: %{gem_name}
```
comps/ has a rubygem- prefix, which is what the packaging guidelines would need too, it should be here?

domcleal commented on rubygem-foreman_scap_client/foreman_scap_client.spec:
```diff
@@ -0,0 +1,90 @@
+URL: https://github.com/openscap/foreman_scap_client
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+Requires: ruby(rubygems)
+%if 0%{?rhel} >= 5 && 0%{?rhel} <= 6
```
`if %{?el6}`

ares replied:
is it equivalent to `el5 || el6`?

domcleal replied:
Well, I don't think this spec file will work at all on EL5...

domcleal commented on rubygem-foreman_scap_client/foreman_scap_client.spec:
```diff
@@ -0,0 +1,90 @@
+%files
+%dir %{gem_instdir}
+%{_bindir}/foreman_scap_client
+%{config_dir}
```
anything need marking as %config in here?

ares replied:
I don't think so, we just create empty dir, we don't provide default config file (just example marged as doc)

ares commented:
@domcleal do you want to give it final review before merge or should I merge and start building?

domcleal commented on comps/comps-foreman-plugins-fedora19.xml:
```diff
@@ -55,14 +58,17 @@
       <packagereq type="default">rubygem-diffy</packagereq>
       <packagereq type="default">rubygem-docker-api</packagereq>
       <packagereq type="default">rubygem-dynflow</packagereq>
+      <packagereq type="default">rubygem-ffi</packagereq>
```
Shouldn't be built for Fedora, it's shipped there.

domcleal commented on rel-eng/tito.props:
```diff
@@ -336,8 +348,10 @@ whitelist = rubygem-algebrick
   rubygem-diffy
   rubygem-docker-api
   rubygem-dynflow
+  rubygem-ffi
```
ditto, already in F19?

domcleal commented:
We should merge and build it bit by bit - seems that ffi needs to be merged for rubygem-openscap builds to complete, and I'll probably need to retag some things so they're mashed correctly.  foreman_openscap should also not be merged until the build deps are here and we can check tests pass.

ares commented:
I squashed in some hacks for foreman_openscap but I still encounter asset precompilation issue, trace at http://pastie.org/9851861

If we need [engine_name fix](https://github.com/OpenSCAP/foreman_openscap/commit/88dced16502438079ad7b62202f976fe1c15fd0a) I can create a 0.2.x branch and build 0.2.1 with this commit tomorrow so we can package it.

domcleal commented:
The engine_name fix looks likely in my opinion.  Does the plugin precompile rake task work on a source installation?

domcleal commented:
Status: everything except foreman_openscap is built, which is waiting on 1.2.1 to include the engine_name fix.

Our repos are currently failing repoclosure as rubygem-openscap depends on openscap >= 1.2.1, which is unavailable in Fedora 19 or the copr repo, so it needs building in one of them, or the dep relaxing.

ares commented:
Status update 
* openscap [built](https://copr.fedoraproject.org/coprs/isimluk/OpenSCAP/build/69047/) in copr repo
* foreman_openscap 0.2.1 [built](http://koji.katello.org/koji/search?match=glob&type=package&terms=*foreman_openscap*) in koji

which means we built everything from this PR so I'm closing it. Thanks @domcleal for all help in this PR.

domcleal commented:
Thanks @ares, great work!
